cips/cip-add-hyperlane.md

```diff
@@ -0,0 +1,53 @@
+| cip           | xx                                                                                         |
+|---------------|--------------------------------------------------------------------------------------------|
+| title         | Add Hyperlane to Celestia                                                                  |
+| description   | This CIP proposes adding Hyperlane to celestia-app.                                        |
+| author        | c-node [@c-node](https://github.com/S1nus), Callum Waters (@cmwaters)                      |
+| discussions-to | [forum](https://forum.celestia.org/t/cip-add-hyperlane-bridging/1909)                     |                                                                     |
+| status        | Draft                                                                                      |
+| type          | Standards Track                                                                            |
+| category      | Core                                                                                       |
+| created       | 2025-02-21                                                                                 |
+
+## Abstract
+
+Add Hyperlane interoperability protocol to the Celestia Network to improve accessibility and usage of TIA
+
+## Motivation
+
+Hyperlane is a popular bridging protocol that has support for EVM, currently the most popular VM for Celestia Rollups, as well as several others, creating connections between some 130 networks. Hyperlane is already employed indirectly to transfer TIA around but is dependent on third party networks, expanding the trust assumptions. Adding Hyperlane natively allows for direct transferring of TIA and possibly other tokens in the future without expanding the trust assumptions beyond the DA network.
+
+## Specification
+
+The full specification of the Hyperlane protocol can be found in their [docs](https://docs.hyperlane.xyz/). The protocol has many similarities with IBC. Messages can be sent and received by writing them to the state tree. A relayer reads the state of one network and relays the messages along with some proof to another network. Warp routes is the standard application built on top of the messaging protocol that supports transferring of tokens.
+
+### Detailed Design
+
+Hyperlane on Celestia, more concretely, requires the addition of two SDK modules: `x/core` and `x/warp`. These are imported from https://github.com/bcp-innovations/hyperlane-cosmos. Token transfer will be enabled upon upgrade.
```

Comment:
[optional] 

```suggestion
Hyperlane on Celestia, more concretely, requires the addition of two SDK modules: `x/core` and `x/warp`. These are imported from https://github.com/bcp-innovations/hyperlane-cosmos. These two modules will be added in the next upgrade which will enable token transfers to other Hyperlane enabled chains.
```

Reply:
The last sentence refers to an actual parameter that needs to be set to actually switch on token transfers. Maybe that wasn't clear